### ISSUE TITLE
keyval_double_free_win.c mixes Comm and Win

### ISSUE THREAD

**Issue by jeffhammond** [pmodels/mpich#6416]
This is a trivial fix.  I'll work on the PR later.

```c
int delete_fn(MPI_Comm comm, int keyval, void *attr, void *extra);
int delete_fn(MPI_Comm comm, int keyval, void *attr, void *extra)
{
    MPI_Win_free_keyval(&keyval);
    return MPI_SUCCESS;
}

...
    MPI_Win_create_keyval(MPI_WIN_NULL_COPY_FN, delete_fn, &keyval, NULL);
```

The same error occurs in keyval_double_free_type.c with `MPI_Datatype`.

**Comment by hzhou** [comment]
Is this covered in https://github.com/pmodels/mpich/pull/6390/commits/0e85e38bc42d1990f01f8640f5e05c91f50e1348?
There are a bunch of fixes in https://github.com/pmodels/mpich/pull/6414

**Comment by jeffhammond** [comment]
Yes, https://github.com/pmodels/mpich/commit/0e85e38bc42d1990f01f8640f5e05c91f50e1348 is the fix.  Thanks.

**Comment by hzhou** [comment]
Fixed by #6414
